### PR TITLE
[cdc] misc. waivers

### PR DESCRIPTION
**Pull request by jeoongp**
AST clocks : waived S_GENCLK errors
SPI_DEVICE :
- Black-boxing dual port SRAM
- Waived paths from TLUL error to u_reg

Signed-off-by: Joshua Park <jeoong@google.com>

### CODE REVIEW

File: hw/cdc/tools/verixcdc/run-cdc.tcl [hunk 1]
```diff
@@ -69,12 +69,13 @@ if {$DEFINE != ""} {
   analyze -sverilog  +define+AST_BYPASS_CLK -f ${SV_FLIST}
 }
 
+# prim_generic_ram_2p is not recognized as a CDC module by the tool
+# So, it's black-boxed while waiting for a tool patch
```
Comment (msfschaffner) [inline hw/cdc/tools/verixcdc/run-cdc.tcl:73]: do we have an issue to capture and track this?
if yes, a common format that we use is `# TODO(# <issue nr>): Short description`

File: hw/cdc/tools/verixcdc/run-cdc.tcl [hunk 2]
```diff
@@ -117,6 +118,9 @@ set_ignore_cdc_paths -name async_fifo_to_ibex_ecc_err -through_signal $async_fif
 set_ignore_cdc_paths -name async_fifo_to_ibex_ivalid -through_signal $async_fifo_data -through_signal [get_pins top_earlgrey/u_rv_core_ibex/u_core/instr_rvalid_i]
 set_ignore_cdc_paths -name async_fifo_to_ibex_dvalid -through_signal $async_fifo_data -through_signal [get_pins top_earlgrey/u_rv_core_ibex/u_core/data_rvalid_i]
 
+# CDC between tlul_fifo_async and regs : ignored
+set tlul_async_data [get_pins -of_objects [get_cells -hier * -filter {ref_name == tlul_fifo_async}] -filter {name =~ tl_d_o*}]
+set_ignore_cdc_paths -name tlul_async_fifo_err -through_signal $tlul_async_data -through_signal [get_pins {top_earlgrey/*/u_reg/u_io_*meas_ctrl_shadowed_cdc/src_we_i}]
```
Comment (msfschaffner) [inline hw/cdc/tools/verixcdc/run-cdc.tcl:123]: Hm... is this really a common synchronizer primitive?

`*meas_ctrl_shadowed_cdc` only seems to be present inside the clock manager...
should we move this waiver to the associated block leve waiver instead?

File: hw/top_earlgrey/cdc/cdc_waivers.tcl [hunk 3]
```diff
@@ -92,3 +92,7 @@ set_rule_status -rule {W_CNTL} -status {Waived} \
 set_rule_status -rule {W_DATA} -status {Waived} \
   -expression {(Signal=~"*u_pinmux_aon.dio_pad_attr_*")} \
   -comment {PAD Attributes are static signals.}
+
+set_rule_status -rule {S_GENCLK} -status {Waived} \
+  -expression {(ClockTreeSignal=~"u_ast.u_ast_clks_byp.*") && (DrivenFlop =~ "u_ast.u_ast_clks_byp.*")} \
+  -comment {ast clks are assumed to be checked by the partner as standalone.}
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.tcl:98]: this is probably ok, assuming that Nuvoton checks for CDC issues at the blocklevel.

**Review by msfschaffner — COMMENTED** [review]
Thanks @jeoongp! I just have a few questions.

File: hw/cdc/tools/verixcdc/run-cdc.tcl (hunk 2 above)
Comment (tjaychen) [reply hw/cdc/tools/verixcdc/run-cdc.tcl:123]: this particular one might become more common.. it's technically applicable to any block that uses the async reg construct.
Do you not see more like this in your setup @jeoongp ?

File: hw/cdc/tools/verixcdc/run-cdc.tcl (hunk 1 above)
Comment (jeoongp) [reply hw/cdc/tools/verixcdc/run-cdc.tcl:74]: Fixed. Thanks

File: hw/top_earlgrey/cdc/cdc_waivers.tcl (hunk 3 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.tcl:98]: I think these errors are caused by not defined in SDC. We need to check if any more RTL/SDC update in AST block later.

**Comment by eunchan** [comment]
LGTM. Please squash two commits or split S_GENCLK, and SPI block-box into two commits.

**Comment by jeoongp** [comment]
> LGTM. Please squash two commits or split S_GENCLK, and SPI block-box into two commits.

Fixed. Thanks

**Review by eunchan — APPROVED** [review]

**Pull request merged**